File: wfm-develop mobile 07.05.2026/wfm/mobile/ios/WFMApp/Features/ManagerHomeFlow/Views/ManagerTaskCardView.swift
import SwiftUI
import WFMUI

/// Карточка задачи для менеджера (отображается на главной странице в секции "Задачи на проверку")
/// Дизайн: Figma node-id=3601:15750
struct ManagerTaskCardView: View {
    let task: Task
    let onTap: () -> Void

    @Environment(\.wfmColors) private var colors

    var body: some View {
        Button(action: onTap) {
            VStack(alignment: .leading, spacing: 0) {
                // Контейнер для badge + название
                VStack(alignment: .leading, spacing: WFMSpacing.xxs) {
                    // Badge категории (только если есть)
                    if let badgeText = task.categoryBadgeText() {
                        WFMBadge(
                            text: badgeText,
                            color: task.categoryBadgeColor()
                        )
                    }

                    // Название зоны
                    Text(task.zoneDisplayName())
                        .wfmHeadline14Medium()
                        .foregroundStyle(colors.cardTextPrimary)
                        .lineLimit(1)
                        .truncationMode(.tail)
                }
                .padding(.top, WFMSpacing.m)
                .padding(.horizontal, WFMSpacing.m)
                .padding(.bottom, WFMSpacing.s)

                // Строка: Время + Имя работника
                HStack(spacing: WFMSpacing.xxxs) {
                    // Иконка часов + время
                    HStack(spacing: WFMSpacing.xxxs) {
                        Image("ic-time")
                            .resizable()
                            .frame(width: 12, height: 12)
                            .foregroundStyle(colors.cardTextTertuary)

                        Text(task.formattedTimeRange())
                            .font(WFMTypography.caption12Regular)
                            .foregroundStyle(colors.cardTextTertuary)
                    }

                    Spacer()

                    // Имя работника
                    if let assignee = task.assignee {
                        Text(assignee.formattedName)
                            .font(WFMTypography.headline14Medium)
                            .foregroundStyle(colors.cardTextPrimary)
                            .lineLimit(1)
                            .truncationMode(.tail)
                    }
                }
                .padding(.horizontal, WFMSpacing.m)
                .padding(.bottom, WFMSpacing.m)
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(colors.cardSurfaceSecondary)
            .overlay(
                RoundedRectangle(cornerRadius: WFMRadius.xl)
                    .stroke(colors.cardBorderSecondary, lineWidth: 1)
            )
            .clipShape(RoundedRectangle(cornerRadius: WFMRadius.xl))
        }
        .buttonStyle(.plain)
    }
}

// MARK: - Preview

#Preview("Manager Task Card") {
    VStack(spacing: 8) {
        ManagerTaskCardView(
            task: Task(
                id: UUID(),
                title: nil,
                description: "Уборка в отделе ФРОВ",
                plannedMinutes: 30,
                assigneeId: 123,
                assignee: AssigneeBrief(
                    id: 123,
                    firstName: "Анна",
                    lastName: "Елисеева",
                    middleName: "Михайловна"
                ),
                state: .completed,
                workType: WorkType(id: 4, name: "Другие работы"),
                zone: Zone(id: 1, name: "ФРОВ", priority: 1),
                category: Category(id: 1, name: "Уборка"),
                timeStart: "08:30:00",
                timeEnd: "09:00:00"
            ),
            onTap: {}
        )

        ManagerTaskCardView(
            task: Task(
                id: UUID(),
                title: nil,
                description: "Молочные продукты",
                plannedMinutes: 60,
                assigneeId: 456,
                assignee: AssigneeBrief(
                    id: 456,
                    firstName: "Иван",
                    lastName: "Петров",
                    middleName: nil
                ),
                state: .completed,
                workType: WorkType(id: 3, name: "Смена ценников"),
                zone: Zone(id: 2, name: "Молочка", priority: 2),
                category: Category(id: 2, name: "Переоценка"),
                timeStart: "11:00:00",
                timeEnd: "12:00:00"
            ),
            onTap: {}
        )
    }
    .padding(16)
    .background(WFMPrimitiveColors.neutral100)
    .wfmTheme()
}

File: wfm-develop mobile 07.05.2026/wfm/mobile/android/app/src/main/java/com/beyondviolet/wfm/features/managerhome/components/ManagerTaskCardView.kt
package com.beyondviolet.wfm.features.managerhome.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.beyondviolet.wfm.R
import com.beyondviolet.wfm.core.models.*
import com.beyondviolet.wfm.ui.components.WfmBadge
import com.beyondviolet.wfm.ui.theme.*
import com.beyondviolet.wfm.core.utils.clickableDebounced

/**
 * Карточка задачи для менеджера (отображается на главной странице в секции "Задачи на проверку")
 * Дизайн: Figma node-id=3601:15750
 */
@Composable
fun ManagerTaskCardView(
    task: Task,
    onTap: () -> Unit,
    modifier: Modifier = Modifier
) {
    val colors = WfmTheme.colors

    Column(
        modifier = modifier
            .fillMaxWidth()
            .background(
                color = colors.cardSurfaceSecondary,
                shape = RoundedCornerShape(WfmRadius.XL)
            )
            .border(
                width = 1.dp,
                color = colors.cardBorderSecondary,
                shape = RoundedCornerShape(WfmRadius.XL)
            )
            .clickableDebounced(debounceTime = 500L, onClick = onTap)
    ) {
        // Контейнер для badge + название
        Column(
            modifier = Modifier.padding(
                top = WfmSpacing.M,
                start = WfmSpacing.M,
                end = WfmSpacing.M,
                bottom = WfmSpacing.S
            ),
            verticalArrangement = Arrangement.spacedBy(WfmSpacing.XXS)
        ) {
            // Badge категории (только если есть)
            task.categoryBadgeText()?.let { badgeText ->
                WfmBadge(
                    text = badgeText,
                    color = task.categoryBadgeScheme()
                )
            }

            // Название зоны
            Text(
                text = task.zoneDisplayName(),
                style = WfmTypography.Headline14Medium,
                color = colors.cardTextPrimary,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis
            )
        }

        // Строка: Время + Имя работника
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(
                    start = WfmSpacing.M,
                    end = WfmSpacing.M,
                    bottom = WfmSpacing.M
                ),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Иконка часов + время
            Row(
                horizontalArrangement = Arrangement.spacedBy(WfmSpacing.XXXS),
                verticalAlignment = Alignment.CenterVertically,
                modifier = Modifier.weight(1f, fill = false)
            ) {
                Icon(
                    painter = painterResource(id = R.drawable.ic_time),
                    contentDescription = null,
                    modifier = Modifier.size(12.dp),
                    tint = colors.cardTextTertuary
                )
                Text(
                    text = task.formattedTimeRange(),
                    style = WfmTypography.Caption12Regular,
                    color = colors.cardTextTertuary
                )
            }

            Spacer(modifier = Modifier.width(WfmSpacing.XXS))

            // Имя работника
            task.assignee?.let { assignee ->
                Text(
                    text = assignee.formattedName(),
                    style = WfmTypography.Headline14Medium,
                    color = colors.cardTextPrimary,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis
                )
            }
        }
    }
}

// MARK: - Preview Helpers

private fun createPreviewTask(
    workTypeId: Int = 4,
    workTypeName: String = "Другие работы",
    assigneeId: Int = 123,
    assignee: AssigneeBrief? = AssigneeBrief(
        id = 123,
        firstName = "Анна",
        lastName = "Елисеева",
        middleName = "Михайловна"
    )
): Task {
    val now = kotlinx.datetime.Clock.System.now()
    return Task(
        id = "preview-manager-task-123",
        title = null,
        description = "Уборка в отделе ФРОВ",
        plannedMinutes = 30,
        assigneeId = assigneeId,
        assignee = assignee,
        state = TaskState.COMPLETED,
        createdAt = now,
        updatedAt = now,
        workTypeId = workTypeId,
        workType = WorkType(id = workTypeId, name = workTypeName),
        zoneId = 1,
        zone = Zone(id = 1, name = "ФРОВ", priority = 1),
        categoryId = 1,
        category = Category(id = 1, name = "Уборка"),
        timeStart = "08:30:00",
        timeEnd = "09:00:00"
    )
}

// MARK: - Previews

@Preview(name = "Manager Task Card - Pink", showBackground = true)
@Composable
private fun ManagerTaskCardViewPreview() {
    WfmTheme {
        Surface(color = WfmTheme.colors.surfaceBase) {
            ManagerTaskCardView(
                task = createPreviewTask(
                    workTypeId = 4,
                    workTypeName = "Другие работы"
                ),
                onTap = {},
                modifier = Modifier.padding(16.dp)
            )
        }
    }
}

@Preview(name = "Manager Task Card - Yellow", showBackground = true)
@Composable
private fun ManagerTaskCardViewYellowPreview() {
    WfmTheme {
        Surface(color = WfmTheme.colors.surfaceBase) {
            ManagerTaskCardView(
                task = createPreviewTask(
                    workTypeId = 3,
                    workTypeName = "Смена ценников",
                    assigneeId = 456,
                    assignee = AssigneeBrief(
                        id = 456,
                        firstName = "Иван",
                        lastName = "Петров",
                        middleName = null
                    )
                ).copy(
                    description = "Молочные продукты",
                    plannedMinutes = 60,
                    zone = Zone(id = 2, name = "Молочка", priority = 2),
                    timeStart = "11:00:00",
                    timeEnd = "12:00:00"
                ),
                onTap = {},
                modifier = Modifier.padding(16.dp)
            )
        }
    }
}

@Preview(name = "Manager Task Card - Multiple", showBackground = true)
@Composable
private fun ManagerTaskCardViewMultiplePreview() {
    WfmTheme {
        Surface(color = WfmTheme.colors.surfaceBase) {
            Column(
                modifier = Modifier.padding(16.dp),
                verticalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                ManagerTaskCardView(
                    task = createPreviewTask(
                        workTypeId = 4,
                        workTypeName = "Другие работы"
                    ),
                    onTap = {}
                )
                ManagerTaskCardView(
                    task = createPreviewTask(
                        workTypeId = 3,
                        workTypeName = "Смена ценников",
                        assigneeId = 456,
                        assignee = AssigneeBrief(
                            id = 456,
                            firstName = "Иван",
                            lastName = "Петров",
                            middleName = null
                        )
                    ).copy(
                        description = "Молочные продукты",
                        plannedMinutes = 60,
                        zone = Zone(id = 2, name = "Молочка", priority = 2),
                        timeStart = "11:00:00",
                        timeEnd = "12:00:00"
                    ),
                    onTap = {}
                )
            }
        }
    }
}
